perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -501,278 +433,245 @@ def _DistributionToBackendFormat(dist):
-          '--num_streams=%s' % FLAGS.object_storage_multistream_num_streams,
-          '--start_time=%s' % read_start_time,
-          '--objects_written_file=%s' % objects_written_file,
-          '--scenario=MultiStreamRead'])
-      try:
-        read_out, _ = vm.RobustRemoteCommand(
-            multi_stream_read_cmd, should_log=True)
-        _ProcessMultiStreamResults(read_out, 'download',
-                                   size_distribution.iterkeys(), results,
-                                   metadata=metadata)
-      except Exception as ex:
-        logging.info('MultiStreamRead test failed with exception %s. Still '
-                     'recording write data.', ex.msg)
-
-      logging.info('Finished multi-stream read test.')
-
-
-def DeleteBucketWithRetry(vm, remove_content_cmd, remove_bucket_cmd):
-  """ Delete a bucket and all its contents robustly.
-
-      First we try to recursively delete its content with retries, if failed,
-      we raise the error. If successful, we move on to remove the empty bucket.
-      Due to eventual consistency issues, some provider may still think the
-      bucket is not empty, so we will add a few more retries when we attempt to
-      remove the empty bucket.
-
-      Args:
-        vm: the vm to run the command.
-        remove_content_cmd: the command line to run to remove objects in the
-            bucket.
-        remove_bucket_cmd: the command line to run to remove the empty bucket.
-
-      Raises:
-        BucketRemovalError: when we failed multiple times to remove the content
-            or the bucket itself.
-  """
-  retry_limit = 0
-  for cmd in [remove_content_cmd, remove_bucket_cmd]:
-    if cmd is remove_content_cmd:
-      retry_limit = CONTENT_REMOVAL_RETRY_LIMIT
-    else:
-      retry_limit = BUCKET_REMOVAL_RETRY_LIMIT
-
-    removal_successful = False
-    logging.info('Performing removal action, cmd is %s', cmd)
-    for i in range(retry_limit):
-      try:
-          vm.RemoteCommand(cmd)
-          removal_successful = True
-          logging.info('Successfully performed the removal operation.')
-          break
-      except Exception as e:
-        logging.error('Failed to perform the removal op. Number '
-                      'of attempts: %d. Error is %s', i + 1, e)
-        time.sleep(RETRY_WAIT_INTERVAL_SECONDS)
-        pass
-
-    if not removal_successful:
-      if cmd is remove_content_cmd:
-        logging.error('Exceeded max retry limit for removing the content of '
-                      'bucket. But we will try to delete the bucket anyway.')
+
+    cmd_parts = [
+        self.test_script_path,
+        '--storage_provider=%s' % self.storage
+    ] + args + self.service.APIScriptArgs()
+    if FLAGS.object_storage_storage_class is not None:
+      cmd_parts += ['--object_storage_class',
+                    FLAGS.object_storage_storage_class]
+
+    return ' '.join(cmd_parts)
+
+# The benchmark functions below all have the same arguments:
```

Comment:
It looks cleaner to have documents inline IMO. Jumping back and forth harms readability and pydoc probably won't work.